### ISSUE TITLE
Allow skills to view more metadata

### ISSUE THREAD

**Issue by manthanhd** [manthanhd/talkify#4]
I want to see following in request object:
```javascript
sentences[] // array of sentences
currentSentence  // pointer to current sentence
currentSentenceIndex  // index of the current sentence

contexts[]  // array of current and previous context states
currentContext  // pointer to current state of context
currentContextIndex  // index of the current context

topics[]  // array of topics
currentTopic  // Pointer to current topic
currentTopicIndex  // index of the current topic
```

Not sure how best these things should be organised.

**Comment by manthanhd** [comment]
Need to refine more information about the `context[]` storage. Do we only save the final context[] or do we save all of them? What's the limit? That could be configurable in the config with a arbitrary number being the default (20 per Id?).

**Comment by manthanhd** [comment]
Current version of this makes sentence and skill (replaces topic) values to be available as part of the request metadata. Closed by #9.

Can't justify recording previous context values at the moment as currently it works by reference. Will consider when another issue is raised or this is reopened.
